accepted/0000-buildpack-yml.md

```diff
@@ -0,0 +1,130 @@
+
+### Decision Points
+
+#### Should all keys be listed under a language family?
+It logically groups configuration for a buildpack family and reduces top level key pollution.
+
+
+## Implementation
+
+Below, we have completed a survey of the existing `buildpack.yml` properties, aligned to the schema, for each language-family.
+
+### Language Families
+
+#### Node.js
+```yaml
+nodejs:
+  engine:
+    version: <semver>
+    optimized-memory: <bool>
+  npm: #we don't support this at all
+    version: <semver>
+  yarn: # unsupported right now
+    version: <semver>
+```
+
+#### .Net Core
+```yaml
+dotnet-core:
+  sdk:
+    version: <semver>
+  aspnet:
+    version: <semver>
+  runtime:
+    version: <semver>
+  build:
+    project-path: <string> # currently supported might need to change
+    projects: #listing of targets to be built during a dotnet core build
+    - <string>
+    - <string>
+```
+
+#### PHP
+```yaml
+php:
+  dist:
+    version: <semver>
+  web: # currently unsupported, all keys underneat this are also unsupported.
+    directory: <string> #currently under the php key as 'webdirectory'
+    script: <string> 
+    libdirectory: <string>
+    serveradmin: <string>
+    enable_https_redirect: <bool>
+    redis:
+      session_store_service_name: <string>
+    memcached:
+      session_store_service_name: <string>
+  web-server: <string> # REMOVE: should be deduced by the presense of nginx or httpd
+```
+
+#### Go
+```yaml
+go: 
+  compiler: # go refers to the `go` binary as the go distribution, might be worth renaming
+    version: <semver>
+  build: #implies a change in go family architecture, here a final go-build buildpack actually builds the binary.
+    import-path: <string>
+    ldflags:
+      <string>: <string>
+      <string>: <string>
+    targets:
+    - <string>
+    - <string>
+```
+
+#### Ruby
+``` yaml
+ruby: # net new, currently unsupported.
+  mri:
+    version: <semver>
+  bundler:
+    version: <semver>
+```
+
+#### Web Server
+```yaml
+web-server: # net new top level key, currently unsupported
+  nginx:
+    version: <semver>
+  httpd:
+    version: <semver>
+```
+
+### Steps
+
+1. Language families should be reviewed to identify all properties that will need to be realigned.
+2. Issues or PRs should be opened to address moving each property, taking into account that the property might need to be "deprecated" so that users will not have a terrible experience once the property moves to its new location.
+3. Once all properties have been realigned for a language family, properties should be documented and outlined for users and buildpack authors.
+
+## Unresolved Questions and Bikeshedding
+
+1. Is `buildpack.yml` the right name?
+
```

Comment:
Would love to see this named something else. config yaml?

Reply:
I don't mind the current name. It specific to what is contained in the file & it's unlikely to collide with anything that might already exist in a user's application.